app/models/ncr/work_order.rb

```diff
@@ -24,21 +26,81 @@ class WorkOrder < ActiveRecord::Base
     validates :building_number, presence: true
     validates :office, presence: true
 
-    # Ignore values in these fields, depending on the expense type
-    def ignore_fields
-      case self.expense_type
+    def set_defaults
+      self.not_to_exceed ||= false
+      self.emergency ||= false
+    end
+
+    # @todo - this is an awkward dance due to the lingering Cart model. Remove
+    # that dependence
+    def create_cart(approver_email, description, requester)
+      cart = Cart.create(
+        name: description,
+        proposal_attributes: {flow: 'linear', client_data: self}
+      )
+      cart.set_requester(requester)
+      self.add_approvals_on(cart, approver_email)
+      Dispatcher.deliver_new_cart_emails(cart)
+      cart
+    end
+    def update_cart(approver_email, description, cart)
+      cart.name = description
+      cart.approver_approvals.destroy_all
+      self.add_approvals_on(cart, approver_email)
+      cart.restart!
+      cart
+    end
+
+    def system_approvers
+      emails = [ENV['NCR_BUDGET_APPROVER_EMAIL'] ||
+                'communicart.budget.approver@gmail.com']
+      if self.expense_type == 'BA61'
+        emails << (ENV['NCR_FINANCE_APPROVER_EMAIL'] ||
+                   'communicart.ofm.approver@gmail.com')
+      else
+        emails
+      end
+    end
+
+    def add_approvals_on(cart, approver_email)
+      emails = [approver_email] + self.system_approvers
+      if self.emergency
+        emails.each {|email| cart.add_observer(email) }
+        # skip state machine
+        cart.proposal.update_attribute(:status, 'approved')
+      else
+        emails.each {|email| cart.add_approver(email) }
+      end
+    end
+
+    # Ignore values in certain fields if they aren't relevant. May want to
+    # split these into different models
+    def self.relevant_fields(expense_type)
+      fields = [:amount, :expense_type, :vendor, :not_to_exceed,
+                :building_number, :office]
+      case expense_type
       when "BA61"
-        ["rwa_number"]
+        fields + [:emergency]
       when "BA80"
-        ["emergency"]
+        fields + [:rwa_number]
       end
     end
 
     # Methods for Client Data interface
     def fields_for_display
-      attributes = self.attribute_names - ["id"] - self.ignore_fields
+      attributes = Ncr::WorkOrder.relevant_fields(self.expense_type)
```

Comment:
Minor: mind moving this into a `#relevant_fields` instance method that calls the class method?